core/templates/pages/exploration-player-page/learner-experience/conversation-skin.directive.ts

```diff
@@ -1238,6 +1252,14 @@ angular.module('oppia').directive('conversationSkin', [
             CurrentInteractionService.submitAnswer();
           };
 
+          $scope.signIn = function() {
```

Comment:
So, this occurs after the first chapter is completed, right? So, is the user's first chapter progress autosaved?

Reply:
Yep! On successful login, the chapter progress will be saved and then the user is redirected to the next chapter.

https://github.com/oppia/oppia/blob/fca6f514a75a5422ce7e0009f51e218dd5d57655/core/controllers/story_viewer.py#L148-L168